app/web/src/features/search/SearchBox.test.tsx

```diff
@@ -68,31 +90,36 @@ describe("SearchBox", () => {
         initialRouterEntries: ["?query=default+value"],
       }).wrapper,
     });
-    const input = screen.getByLabelText(USER_SEARCH);
+    const input = screen.getByLabelText(PROFILE_KEYWORDS);
     expect(input).toHaveValue("default value");
     userEvent.click(screen.getByRole("button", { name: CLEAR_SEARCH }));
     await waitFor(() => {
       expect(input).toHaveValue("");
     });
   });
 
-  it("opens and closes the filter dialog", async () => {
+  it("opens and closes the filter dialog, with changes applied to search box", async () => {
     const setActive = jest.fn();
     render(<View setActive={setActive} />, { wrapper });
-    const input = screen.getByLabelText(USER_SEARCH);
+    userEvent.click(screen.getByLabelText(SEARCH_BY_KEYWORD));
+    const input = screen.getByLabelText(PROFILE_KEYWORDS);
     userEvent.type(input, "test search");
-    userEvent.click(screen.getByRole("button", { name: OPEN_FILTER_DIALOG }));
+    await waitFor(() => {
+      expect(setActive).toBeCalledWith({ query: "test search" });
+    });
+    userEvent.click(screen.getByRole("button", { name: FILTER_DIALOG_TITLE }));
 
     const dialog = screen.getByRole("dialog", { name: FILTER_DIALOG_TITLE });
     expect(dialog).toBeVisible();
+    const dialogKeywordsField = within(dialog).getByLabelText(PROFILE_KEYWORDS);
+    expect(dialogKeywordsField).toHaveValue("test search");
+    userEvent.clear(dialogKeywordsField);
+    userEvent.type(dialogKeywordsField, "new search");
     userEvent.click(screen.getByRole("button", { name: APPLY_FILTER }));
     await waitFor(() => {
       expect(dialog).not.toBeVisible();
     });
 
-    expect(input).toHaveValue("test search");
-    await waitFor(() => {
-      expect(setActive).toBeCalledWith({ query: "test search" });
-    });
+    expect(input).toHaveValue("new search");
   });
 });
```

Comment:
Missing tests to check:
- clearing the location from the input clears filter
- clearing the keyword search from input clears filter